client/components/contact/ContactForm.jsx

```diff
@@ -0,0 +1,270 @@
+    setFormValues({
+      ...initialFormValues,
+    });
+  }
+
+  // initialize component
+  useEffect(() => {
+    // componentDidMount code goes here...
+    clearFields();
+    if (!!displayFeedbackSuccess === true) {
+      toast.success('We received your message. Our team will contact you at the email address provided.', {
+        position: 'top-right',
+        autoClose: 5000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+      });
+    }
+
+    if (!!openErrorModal === true) {
+      toast.error('We failed to process your message. Please try again later.', {
+        position: 'top-right',
+        autoClose: 5000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+      });
+    }
+
+    return () => {
+      // componentWillUnmount code goes here...
+      callShowFeedbackSuccess(false);
+      callShowErrorModal(false);
+      clearFields();
+    };
+  }, [callShowErrorModal, callShowFeedbackSuccess, displayFeedbackSuccess, openErrorModal]);
+
+  // helper methods
+  function validateEmail(emailAddress) {
+    // eslint-disable-next-line
+    if (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(emailAddress)) {
+      return true;
+    }
+    return false;
+  }
+
+  const clearErrors = useCallback(() => {
+    setFormValues(prevState => ({
+      ...prevState,
+      ...{
+        errors: {
+          missingFirstName: false,
+          missingLastName: false,
+          missingEmail: false,
+          invalidEmail: false,
+          missingMessage: false,
+        },
+      },
+    }));
+  }, []);
+
+  const validateForm = useCallback(() => {
+    const noFirstName = formValues.firstName.trim().length === 0;
+    const noLastName = formValues.lastName.trim().length === 0;
+    const noEmail = formValues.email.trim().length === 0;
+    const noMessage = formValues.message.trim().length === 0;
+    const incompleteEmail = (!noEmail && !validateEmail(formValues.email));
+    if (!noFirstName && !noLastName && !noEmail && !noMessage && !incompleteEmail) {
+      return true;
+    }
+
+    setFormValues(prevState => ({
+      ...prevState,
+      ...{
+        errors: {
+          missingFirstName: noFirstName,
+          missingLastName: noLastName,
+          missingEmail: noEmail,
+          invalidEmail: incompleteEmail,
+          missingMessage: noMessage,
+        },
+      },
+    }));
+    return false;
+  }, [formValues]);
+
+  // event handlers
+  const onInputChange = useCallback(event => {
+    event.preventDefault();
+    const { name, value } = event.target;
+    setFormValues(prevState => ({ ...prevState, [name]: value }));
+  }, []);
+
+  const handleSubmit = useCallback(event => {
+    event.preventDefault();
+
+    if (validateForm()) {
+      const body = [
+                `First name: ${formValues.firstName}`,
```

Comment:
Do we want to use the trimmed versions here?

Reply:
you are right. i know we're not making any system calls with the input but do you think we need to do any other sort of further input validation/sanitization? like prevent or scrub out "bad characters" like ";", "<" or "&"